app/src/androidTest/java/ch/epfl/balelecbud/util/cache/FilesystemCacheTest.java

```diff
@@ -0,0 +1,159 @@
+package ch.epfl.balelecbud.util.cache;
+
+
+import android.util.Log;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import ch.epfl.balelecbud.BalelecbudApplication;
+import ch.epfl.balelecbud.models.Location;
+import ch.epfl.balelecbud.schedule.models.Slot;
+import ch.epfl.balelecbud.testUtils.TestAsyncUtils;
+import ch.epfl.balelecbud.util.database.Database;
+import ch.epfl.balelecbud.util.database.MockDatabase;
+import ch.epfl.balelecbud.util.database.MyGeoClause;
+import ch.epfl.balelecbud.util.database.MyQuery;
+import ch.epfl.balelecbud.util.database.MyWhereClause;
+
+import static ch.epfl.balelecbud.util.database.Database.DOCUMENT_ID_OPERAND;
+import static ch.epfl.balelecbud.util.database.MockDatabase.assertQueryMapResults;
+import static ch.epfl.balelecbud.util.database.MockDatabase.assertQueryResults;
+import static ch.epfl.balelecbud.util.database.MockDatabase.axel;
+import static ch.epfl.balelecbud.util.database.MockDatabase.karim;
+import static ch.epfl.balelecbud.util.database.MockDatabase.slot1;
+import static ch.epfl.balelecbud.util.database.MockDatabase.slot2;
+import static ch.epfl.balelecbud.util.database.MockDatabase.token1;
+import static ch.epfl.balelecbud.util.database.MyWhereClause.Operator.EQUAL;
+
+@RunWith(AndroidJUnit4.class)
+public class FilesystemCacheTest {
+
+    private Cache appCache = BalelecbudApplication.getAppCache();
+
+    @Before
+    public void setup() {
+        BalelecbudApplication.setRemoteDatabase(MockDatabase.getInstance());
+        appCache.flush();
+    }
+
+    @Test
+    public void containsCorrectForCollectionRetrieval() throws IOException {
```

Comment:
Avoid using "Correct" in test method names, you can rename to testContainsForCollectionRetrieval. Same for line 63